include/deal.II/base/exceptions.h

```diff
@@ -314,6 +319,74 @@ namespace deal_II_exceptions
                               const char       *cond,
                               const char       *exc_name,
                               ExceptionBase     e) noexcept;
+#ifdef DEAL_II_WITH_CUDA
+    /**
+     * Return a string given an error code. This is similar to the cudaGetErrorString
+     * function but there is no equivalent function for cuSPARSE.
+     */
+    inline
+    std::string get_cusparse_error_string(cusparseStatus_t error_code)
```

Comment:
can you declare the function here and move its implementation into the .cc file?

Reply:
Sure, I don't know why I thought that there was no .cc file.